pkg/driver/provisioner_server_impl.go

```diff
@@ -133,6 +134,18 @@ func (s *ProvisionerServer) DriverCreateBucket(ctx context.Context,
 		if errors.As(err, &bucketAlreadyExists) {
 			klog.V(c.LvlInfo).InfoS("Bucket already exists", "bucketName", bucketName)
 			return nil, status.Errorf(codes.AlreadyExists, "Bucket already exists: %s", bucketName)
+		} else if strings.Contains(err.Error(), "InvalidBucketName") {
+			klog.ErrorS(err, "Invalid bucket name - will not retry", "bucketName", bucketName)
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid bucket name: %s", bucketName)
+		} else if strings.Contains(err.Error(), "InvalidLocationConstraint") {
+			klog.ErrorS(err, "Invalid location constraint - will not retry", "bucketName", bucketName)
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid location constraint for bucket: %s", bucketName)
+		} else if strings.Contains(err.Error(), "AccessDenied") {
+			klog.ErrorS(err, "Access denied - will not retry", "bucketName", bucketName)
+			return nil, status.Errorf(codes.PermissionDenied, "Access denied for bucket: %s", bucketName)
+		} else if strings.Contains(err.Error(), "InvalidRequest") || strings.Contains(err.Error(), "MalformedXML") {
+			klog.ErrorS(err, "Invalid request format - will not retry", "bucketName", bucketName)
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid request for bucket: %s", bucketName)
```

Comment:
Can you use a mapping to simplify this like

Reply:
I was thinking its simpler and clearner the original way